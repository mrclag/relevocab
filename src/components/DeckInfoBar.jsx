import React from 'react';
import deckImg from '../images/search_b.png';
import { Link } from 'react-router-dom';
import { deleteDeck } from '../store/actions/deckActions';

import { connect } from 'react-redux';

const DeckInfoBar = ({ currentDeck, deleteDeck }) => {
  return (
    <>
      <div className="top-left">
        <img src={deckImg} alt="" className="deck-picture" />
        <div className="deck-info">
          <div className="deck-info-title">{currentDeck.title}</div>
        </div>
      </div>
      <div className="deck-buttons">
        <Link to="/practice">
          <button className="button button-practice">Practice</button>
        </Link>
        <button
          onClick={() => deleteDeck(currentDeck.deckId)}
          className="button button-delete"
        >
          Remove
        </button>
      </div>
    </>
  );
};

const mapDispatchToProps = dispatch => {
  return {
    deleteDeck: deckId => dispatch(deleteDeck(deckId))
  };
};

export default connect(null, mapDispatchToProps)(React.memo(DeckInfoBar));
