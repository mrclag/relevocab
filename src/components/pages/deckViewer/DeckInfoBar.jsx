import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import { deleteDeck } from '../../../store/actions/deckActions';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faImage } from '@fortawesome/free-solid-svg-icons';

export const DeckInfoBar = ({ currentDeck, deleteDeck, currentCards }) => {
  return (
    <>
      <div className="top-left">
        {/* <img src={file_upload} alt="" className="deck-picture" /> */}
        {currentDeck.imgUrl ? (
          <img src={currentDeck.imgUrl} alt="" className="deck-picture" />
        ) : (
          <FontAwesomeIcon
            icon={faImage}
            className="deck-picture"
            color="#333"
          />
        )}
        <div className="deck-info">
          <div className="deck-info-title">
            {currentDeck && currentDeck.title}
          </div>
          <div>{currentCards && currentCards.length} cards</div>
        </div>
      </div>
      <div className="deck-buttons">
        <Link to="/practice">
          <button className="button button-practice">Practice</button>
        </Link>
        <button
          onClick={() => deleteDeck(currentDeck.deckId)}
          className="button button-delete"
        >
          Remove
        </button>
      </div>
    </>
  );
};

const mapDispatchToProps = (dispatch) => {
  return {
    deleteDeck: (deckId) => dispatch(deleteDeck(deckId)),
  };
};

const mapStateToProps = (state) => {
  return {
    currentCards: state.deck.currentCards,
    currentDeck: state.deck.currentDeck,
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(React.memo(DeckInfoBar));
