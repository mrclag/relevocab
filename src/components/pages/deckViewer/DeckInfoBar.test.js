import React from 'react';
import { DeckInfoBar } from './DeckInfoBar.jsx';
import { shallow } from 'enzyme';

describe('DeckInfoBar', () => {
  const deckInfoBar = shallow(<DeckInfoBar />);

  it('renders correctly', () => {
    expect(deckInfoBar).toMatchSnapshot();
  });
});
