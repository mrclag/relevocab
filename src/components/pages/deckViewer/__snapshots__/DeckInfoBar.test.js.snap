// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeckInfoBar renders correctly 1`] = `ShallowWrapper {}`;
